test/TritonGPU/amd/stream-pipeline-v2.mlir

```diff
@@ -0,0 +1,36 @@
+// RUN: triton-opt %s -tritonamdgpu-stream-pipeline-v2 | FileCheck %s
+
+// Check that the stream pipeliner pass produces correct IR when it has to replace pre-existing local buffers
+// CHECK-LABEL: loop_with_dot_and_transpose
+#blocked = #triton_gpu.blocked<{sizePerThread = [2, 2], threadsPerWarp = [2, 16], warpsPerCTA = [4, 1], order = [1, 0]}>
+#blocked1 = #triton_gpu.blocked<{sizePerThread = [1, 1], threadsPerWarp = [32, 1], warpsPerCTA = [1, 4], order = [0, 1]}>
+#shared = #triton_gpu.shared<{vec = 1, perPhase = 1, maxPhase = 1, order = [0, 1], hasLeadingOffset = false}>
+#shared1 = #triton_gpu.shared<{vec = 1, perPhase = 1, maxPhase = 1, order = [1, 0], hasLeadingOffset = false}>
+module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 4 : i32, triton_gpu.target = "hip:gfx1201", "triton_gpu.threads-per-warp" = 32 : i32} {
+  tt.func public @loop_with_dot_and_transpose(%arg0: !tt.ptr<f32> {tt.divisibility = 16 : i32, tt.pointer_range = 32 : i32}, %arg1: i32) attributes {noinline = false} {
+    %c1_i32 = arith.constant 1 : i32
+    %c0_i32 = arith.constant 0 : i32
+    %cst = arith.constant dense<0.000000e+00> : tensor<32x32xf32, #blocked>
+    %0 = tt.make_range {end = 32 : i32, start = 0 : i32} : tensor<32xi32, #triton_gpu.slice<{dim = 1, parent = #blocked1}>>
+    %1 = tt.expand_dims %0 {axis = 1 : i32} : tensor<32xi32, #triton_gpu.slice<{dim = 1, parent = #blocked1}>> -> tensor<32x1xi32, #blocked1>
+    %2 = tt.splat %arg0 : !tt.ptr<f32> -> tensor<32x1x!tt.ptr<f32>, #blocked1>
+    %3 = tt.addptr %2, %1 : tensor<32x1x!tt.ptr<f32>, #blocked1>, tensor<32x1xi32, #blocked1>
+    %4 = tt.make_range {end = 32 : i32, start = 0 : i32} : tensor<32xi32, #triton_gpu.slice<{dim = 0, parent = #blocked1}>>
+    %5 = tt.expand_dims %4 {axis = 0 : i32} : tensor<32xi32, #triton_gpu.slice<{dim = 0, parent = #blocked1}>> -> tensor<1x32xi32, #blocked1>
+    %6 = tt.broadcast %3 : tensor<32x1x!tt.ptr<f32>, #blocked1> -> tensor<32x32x!tt.ptr<f32>, #blocked1>
+    %7 = tt.broadcast %5 : tensor<1x32xi32, #blocked1> -> tensor<32x32xi32, #blocked1>
+    %8 = tt.addptr %6, %7 : tensor<32x32x!tt.ptr<f32>, #blocked1>, tensor<32x32xi32, #blocked1>
+    %9 = scf.for %arg2 = %c0_i32 to %arg1 step %c1_i32 iter_args(%arg3 = %cst) -> (tensor<32x32xf32, #blocked>)  : i32 {
+      %11 = tt.load %8 : tensor<32x32x!tt.ptr<f32>, #blocked1>
+      %12 = triton_gpu.local_alloc %11 : (tensor<32x32xf32, #blocked1>) -> !tt.memdesc<32x32xf32, #shared, #triton_gpu.shared_memory>
```

Comment:
Can you add a `// CHECK:` line for local_alloc and trans ops?

Reply:
Added check to verify transpose result is `mutable` 
https://github.com/triton-lang/triton/pull/4969/files#diff-1859c7f7dbc311a02b8fa31614bb124ac822496391eed1005aac5d207792d70cR241-R242